doc/model.rst

```diff
@@ -0,0 +1,932 @@
+
+Continue to the :ref:`loading an in-memory TorchScript function<in_mem_TF_ex>` example.
+
+.. _in_mem_TF_ex:
+Example: Loading a in-memory TorchScript function
+-------------------------------------------------
+This example walks through the steps of instructing SmartSim to load an in-memory TorchScript function
+to a standard ``Orchestrator``.
+
+.. note::
+    The example assumes:
+
+    - a standard ``Orchestrator`` is launched prior to the ``Models`` execution
+    - an initialized ``Model`` named `smartsim_model` exists within the ``Experiment`` workflow
+
+**Define an in-memory TF function**
+
+To begin, define an in-memory TorchScript function within the Python driver script.
+For the purpose of the example, we add a simple TorchScript function, `timestwo`:
+
+.. code-block:: python
+
+    def timestwo(x):
+        return 2*x
+
+**Attach the in-memory TorchScript function to a SmartSim Model**
+
+We use the ``Model.add_function()`` function to instruct SmartSim to load the TorchScript function `timestwo`
+onto the launched standard ``Orchestrator``. Specify the function `timestwo` to the `function`
+parameter:
+
+.. code-block:: python
+
+    smartsim_model.add_function(name="example_func", function=timestwo, device="GPU", devices_per_node=2, first_device=0)
+
+In the above ``smartsim_model.add_function()`` code snippet, we offer the following arguments:
+
+-  `name` ("example_func"): A key to uniquely identify the model within the database.
+-  `function` (timestwo): Name of the TorchScript function defined in the Python driver script.
+-  `device` ("CPU"): Specifying the device for ML model execution.
+-  `devices_per_node` (2): Use two GPUs per node.
+-  `first_device` (0): Start with 0 index GPU.
+
+.. warning::
+    Calling `exp.start(smartsim_model)` prior to instantiation of an orchestrator will result in
+    a failed attempt to load the ML model to a non-existent database.
+
+When the ``Model`` is started via ``Experiment.start()``, the TF function will be loaded to the
+standard ``Orchestrator``. The function can then be executed on the ``Orhcestrator`` via a SmartSim
+client (:ref:`SmartRedis<dead_link>`) within the application code.
+
+.. _TS_from_file:
+------------------------------
+Attach a TorchScript from file
+------------------------------
+Users can attach TorchScript functions from a file to a ``Model`` and upload them to a
+colocated or standard ``Orchestrator``. This functionality is supported by the ``Model.add_script()``
+function's `script_path` parameter. The function supports
+flexible device selection, allowing users to choose between `"GPU"` or `"CPU"` via the `device` parameter.
+In environments with multiple devices, specific device numbers can be specified using the
+`devices_per_node` parameter.
+
+When specifying a TorchScript using ``Model.add_script()``, the
+following arguments are offered:
+
+- `name` (str): key to store script under.
```

Comment:
key to store function under.

->

reference name for the script inside of the ``Orchestrator``